### DISCUSSION TITLE
Modernize codebases (TS/Kotlin/Swift)

### DISCUSSION THREAD

**Discussion by birkskyum** [maplibre/maplibre#185] [Cross-project feature proposals]
**iOS / Swift**

Apart from a much better DX, Apple has been very clear that the Swift will replace Obj-C / Obj-C++, and they start to make that very clear for us.

For every new iOS release, APIs change, but where the old API is deprecated in both Obj-C and Swift, often only Swift will contain the new recommended API. For iOS 15.5 we i.e. had to disable a range of unit tests of NSExpression written in Obj-C, and migrating them to Swift will allow us to reactive them because Swift allow for the new syntax. [[Link to open issue](https://github.com/maplibre/maplibre-gl-native/issues/421)]

We can make a gradual migration: https://developer.apple.com/documentation/swift/migrating-your-objective-c-code-to-swift

**Android / Kotlin**

The work on porting Java to Kotlin [started sep. 2022](https://github.com/maplibre/maplibre-gl-native/blob/main/design-proposals/2022-09-02-kotlin.md), and is well under way (~45% complete), but it's on ad-hoc basis and allocation of resources would surely speed things up.

**Web / TypeScript**

This work is complete in the gl-js repo, but there are still a lot of es5 js script in the gl-native repo that needs taken care of.

**Comment by wipfli** [comment]
Thanks for this suggestion. I think it would be best to open an issue in MapLibre GL Native if there is not already one for this issue.

**Reply by birkskyum** [reply]
I have updated this one to match the "Cross-project strategic direction" label. Hope it's alright.
